### ISSUE TITLE
gdb/database: `gdb.Model.Scan` did not return `sql.ErrNoRows` when record not found

### ISSUE THREAD

**Issue by RyoJerryYu** [gogf/gf#4031]
### Go version

go version go1.23.4 linux/amd64

### GoFrame version

v2.7.4

### Can this bug be reproduced with the latest release?

Option Yes

### What did you do?

We use `Scan` as bellow, where the result would be not exist:

```go
var res *SomeModel
err := s.Model(ctx).WherePri(in.Id).Scan(&res)
if err == sql.ErrNoRow {
    // expecting it would go here if record not found, just like other famous golang packages.
}
if err != nil {
    // expecting other error except not found
}

// expecting res should be not nil, since no error return
```

### What did you see happen?

It return no error, and the `res` still remain `nil` .

### What did you expect to see?

![Screenshot from 2024-12-11 11-02-46](https://github.com/user-attachments/assets/31b3354a-ba2d-4248-9724-f27aaf6f48ad)

I found that in the `Struct` method within `database/gdb/gdb_model_select.go`, there is a check that returns `sql.ErrNoRows` when `pointer` is not `nil`. Why doesn't it return `sql.ErrNoRows` when `pointer` is `nil`? Is this by design, or just a bug?

If by design, please tell me why. Since it is scanning a single record into a struct, it is quite reasonable to return a `sql.ErrNoRows`. This practice is also followed by other Go packages, such as `database/sql`, `sqlx`, `jet`, `sqlc`, and others.

Now, the `Scan` operation into a `**struct` does not return `sql.ErrNoRows` and still retains `nil`, which is extremely confusing, especially for those who are already familiar with the GoLang ecosystem.

---

BTW, I've noticed that the current implementation results in different behaviors when `res` is a pointer or a struct, which also contributes to the confusion.

```go
var res *SomeModel // a pointer
err := s.Model(ctx).WherePri(in.Id).Scan(&res)
// return no error, res still nil

// ...

var res SomeModel // a struct allocated
err := s.Model(ctx).WherePri(in.Id).Scan(&res)
// different to the above, it would return a sql.ErrNoRows , so confusing
```

**Comment by gqcn** [comment]
@RyoJerryYu Yes, goframe ORM performs different from other famous golang ORM library. Please refer to: https://goframe.org/en/docs/core/gdb-result-empty-check

**Comment by RyoJerryYu** [comment]
> @RyoJerryYu Yes, goframe ORM performs different from other famous golang ORM library. Please refer to: https://goframe.org/en/docs/core/gdb-result-empty-check


@gqcn 
When the passed object is a null pointer, and no data is found, why not return the `sql.ErrNoRows` ? The reason that 'one can determine if the return is empty by checking if the pointer is nil' is not reasonable enough. Ensuring the interface's compatibility with standard definitions and the stability of its promised behavior is clearly more important.
